processing/data_collection/gazette/spiders/base.py

```diff
@@ -24,6 +27,88 @@ def __init__(self, start_date=None, *args, **kwargs):
             self.logger.info("Collecting all gazettes available")
 
 
+class SigpubGazetteSpider(BaseGazetteSpider):
+    """www.diariomunicipal.com.br (Sigpub) base spider
+
+    Documents obtained by this kind of spider are text-PDFs with many cities in it.
+    That's because the websites are usually made for associations of cities.
+    
+    TODO:
+        - All variations have a "possible" start date of 01/01/2009, but that may cause
+        many unnecessary requests to be made if they actually start making available
+        documents later. Some investigation for the start date of each website needs to
+        be made in this case.
+
+    Observations:
+        - These websites have an "Advanced Search", but they are protected by ReCaptcha.
+    """
+
+    custom_settings = {"USER_AGENT": UserAgent().random}
+    FIRST_AVAILABLE_DATE = date(2009, 1, 1)
+
+    def start_requests(self):
+        """Requests start page where the calendar widget is available."""
+        yield scrapy.Request(self.CALENDAR_URL, callback=self.parse_calendar)
+
+    def parse_calendar(self, response):
+        """Makes requests for each date to see if a document is available."""
+        default_form_fields = {
+            "calendar[_token]": response.xpath(
+                "//input[@id='calendar__token']/@value"
+            ).get()
+        }
+        for date, date_form_fields in self.available_dates_form_fields():
+            formdata = {**default_form_fields, **date_form_fields}
+
+            yield scrapy.FormRequest(
+                url=response.urljoin("materia/calendario"),
+                formdata=formdata,
+                meta={"date": date, "edition_type": "regular"},
+                callback=self.parse_gazette_info,
+            )
+            yield scrapy.FormRequest(
+                url=response.urljoin("materia/calendario/extra"),
+                formdata=formdata,
+                meta={"date": date, "edition_type": "extra"},
+                callback=self.parse_gazette_info,
+            )
+
+    def parse_gazette_info(self, response):
+        """Parses document availability endpoint and gets document URL if available."""
+        body = json.loads(response.text)
+        meta = response.meta
+
+        if "error" in body:
+            self.logger.warning(
+                f"{meta['edition_type'].capitalize()} Gazette not available for {meta['date'].date()}"
+            )
+            return
```

Comment:
I do not think this worth a warning message. To me, it is more like a debug message. Just because a type of edition is not present does not mean there is something wrong or might be something wrong. Right? What do you think?

Reply:
You're right. Changed it.